### PR TITLE
Support several CALYPSO run in different pressure in one iteration

### PR DESCRIPTION
**Pull request by wangzyphysics**
Support several CALYPSO run in different pressure in one iteration:

One can do this by changing `model_devi_jobs` into  
```
{"times":[4, 5],
"NameOfAtoms":["Mg","Al","Cu"],
"NumberOfAtoms":[1,1,1],
"NumberOfFormula":[1,1],
"Volume":30,
...
"PSTRESS":[0, 400, 500, 600],
"fmax":0.01}
```

the `PSTRESS` is a `List` which can contain several different pressure and the number of CALYPSO Loop is same as the length of `PSTRESS` in one iteration

here is picture of this situation

```
$ pwd
/home/zhenyu/workplace/dpgen/debug/iter.000004/01.model_devi
```
image

the `times` is a `List` which can be seen as a collection of  `iter number`, in this example `"times":[4, 5]` means iter.000004 and iter.000005 will run the same CALYPSO Loop

### CODE REVIEW

File: dpgen/generator/lib/make_calypso.py [hunk 1]
```diff
@@ -75,18 +76,18 @@ def make_calypso_input(nameofatoms,numberofatoms,
     ret+= "NumberOfParallel = 4\n"
     assert split != None 
     ret+= "Split = %s\n"%(split)
-    assert pstress != None  or type(pstress) == type([200])
-    ret+= "PSTRESS = %f\n"%(pstress[0])
-    assert fmax != None  or type(fmax) == type([200])
-    ret+= "fmax = %f\n"%(fmax[0])
+    assert pstress != None  or type(pstress) == type(200) or type(pstress) == type(0.001)
```
Comment (wanghan-iapcm) [inline dpgen/generator/lib/make_calypso.py:79]: similar issue

File: dpgen/generator/lib/make_calypso.py [hunk 2]
```diff
@@ -95,18 +96,19 @@ def make_calypso_input(nameofatoms,numberofatoms,
         ret+= "###################End Parameters for VSC ##########################\n"
     return ret
 
-def _make_model_devi_buffet(jdata,calypso_run_opt_path):
+def _make_model_devi_buffet(jdata,caly_run_opt_path):
```
Comment (wanghan-iapcm) [inline dpgen/generator/lib/make_calypso.py:99]: I would prefer the original variable name, which is self-explanatory to the readers. It is not easy to relate `caly` to `calypso`...

File: dpgen/generator/run.py [hunk 3]
```diff
@@ -831,25 +833,43 @@ def make_model_devi (iter_index,
...
-        run_opt_script = os.path.join(calypso_run_opt_path,'calypso_run_opt.py')
-        shutil.copyfile(run_opt_file,run_opt_script)
-        # check outcar script
-        check_outcar_script = os.path.join(calypso_run_opt_path,'check_outcar.py')
-        shutil.copyfile(check_outcar_file,check_outcar_script)
+        # Create work path list
+        caly_run_opt_path = []
+        if run_mode == 1:
+            caly_run_opt_path.append('%s.%03d'%(_calypso_run_opt_path, 0))
+        elif run_mode == 2:
```
Comment (wanghan-iapcm) [inline dpgen/generator/run.py:846]: add a comment explaining what does the run_modes mean?

**Review by wanghan-iapcm — CHANGES_REQUESTED** [review]

File: dpgen/generator/lib/run_calypso.py [hunk 4]
```diff
@@ -329,20 +364,16 @@ def analysis(iter_index,jdata,calypso_run_opt_path,calypso_model_devi_path):
             dlog.info(npos,'failed : ',e)
 
     if len(ms) == 0:
-        print('too little confs, ')
-        return
+        dlog.info('too little confs, ')
+        sys.exit()
```
Comment (njzjz) [inline dpgen/generator/lib/run_calypso.py:368]: Use `raise` instead

File: dpgen/generator/run.py [hunk 5]
```diff
@@ -831,25 +833,65 @@ def make_model_devi (iter_index,
...
         calypso_run_model_devi_script = os.path.join(calypso_model_devi_path,'calypso_run_model_devi.py')
         shutil.copyfile(calypso_run_model_devi_file,calypso_run_model_devi_script)
-        # run confs opt script
-        run_opt_script = os.path.join(calypso_run_opt_path,'calypso_run_opt.py')
-        shutil.copyfile(run_opt_file,run_opt_script)
-        # check outcar script
-        check_outcar_script = os.path.join(calypso_run_opt_path,'check_outcar.py')
-        shutil.copyfile(check_outcar_file,check_outcar_script)
+        # Create work path list
+        caly_run_opt_path = []
```
Comment (wanghan-iapcm) [inline dpgen/generator/run.py:843]:
```suggestion
        calypso_run_opt_path = []
```
To improve the readability of the code

**Review by wanghan-iapcm — APPROVED** [review]

**Comment by wanghan-iapcm** [comment]
Could you please merge the devel branch to fix the bug in UT? Thanks!

**Review by njzjz — APPROVED** [review]

**Pull request merged**